File: frontend/apps/web/hooks/api/types/comment.ts
export interface Comment {
  id: string
  market_id: string
  user_id: string
  username: string
  content: string
  depth: number
  parent_id: string | null
  reply_count: number
  is_deleted: boolean
  created_at: string
  updated_at: string
}

export interface CommentsResponse {
  success: boolean
  data: {
    comments: Comment[]
    page: number
    page_size: number
  }
}

File: backend/app/schemas/comment.py
from datetime import datetime

from pydantic import BaseModel, Field


class CommentCreate(BaseModel):
    content: str = Field(..., min_length=1, max_length=2000)
    parent_id: str | None = None


class CommentResponse(BaseModel):
    id: str
    market_id: str
    user_id: str
    username: str
    parent_id: str | None = None
    content: str
    depth: int
    is_deleted: bool
    reply_count: int
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}
